modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/admin/dashboard/authentication/AdminDashboardLogin.java

```diff
@@ -0,0 +1,88 @@
+*/
+package org.wso2.appmanager.integration.test.cases.admin.dashboard.authentication;
+
+import org.json.JSONObject;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.appmanager.integration.utils.APPMAdminDashboardRestClient;
+import org.wso2.appmanager.integration.utils.AppmTestConstants;
+import org.wso2.carbon.automation.engine.context.AutomationContext;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.carbon.automation.engine.context.beans.User;
+import org.wso2.carbon.automation.test.utils.http.client.HttpResponse;
+
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Test case which verifies the ability of appCreator, appPublisher and admin users of logging to admin dashboard.
+ */
+public class AdminDashboardLogin {
+    private static final String TEST_DESCRIPTION = "Verify login to Admin Dashboard";
+    private String backEndUrl;
+    private static AutomationContext appMServer;
+
+    @BeforeClass(alwaysRun = true)
+    public void startUp() throws Exception {
+        appMServer = new AutomationContext(AppmTestConstants.APP_MANAGER, TestUserMode.SUPER_TENANT_ADMIN);
+        backEndUrl = appMServer.getContextUrls().getWebAppURLHttps();
+    }
+
+    @Test(dataProvider = "validUserModeDataProvider", description = TEST_DESCRIPTION)
+    public void testAdminDashboardLoginWithValidUsers(String userName, String password) throws Exception {
+        APPMAdminDashboardRestClient appmAdminDashboardRestClient = new APPMAdminDashboardRestClient(backEndUrl);
+        HttpResponse response = appmAdminDashboardRestClient.login(userName, password);
+        JSONObject publisherLogoutJsonObject = new JSONObject(response.getData());
+        String dataResponse = publisherLogoutJsonObject.getString(AppmTestConstants.ERROR);
+        assertTrue(dataResponse == "false", "Login to Admin dashboard is not allowed for user : " + userName + " who " +
+                "has enough privileges to login.");
+        String session = response.getHeaders().get(AppmTestConstants.SET_COOKIE);
+        assertNotNull(session, "Session is null");
+        appmAdminDashboardRestClient.logout();
+    }
+
+    @Test(dataProvider = "inValidUserModeDataProvider", description = TEST_DESCRIPTION)
+    public void testAdminDashboardLoginWithInValidUsers(String userName, String password) throws Exception {
+        APPMAdminDashboardRestClient appmAdminDashboardRestClient = new APPMAdminDashboardRestClient(backEndUrl);
+        HttpResponse response = appmAdminDashboardRestClient.login(userName, password);
+        JSONObject publisherLogoutJsonObject = new JSONObject(response.getData());
+        String dataResponse = publisherLogoutJsonObject.getString(AppmTestConstants.ERROR);
+        assertTrue(dataResponse == "true", "Login to Admin dashboard is allowed for user : " + userName + " who " +
```

Comment:
Fixed in https://github.com/wso2/product-app-manager/commit/e263afed698e7dbd1253c0136563513fd4051eea